backend/src/main/java/com/google/sps/servlets/AddParticipantServlet.java

```diff
@@ -56,33 +58,48 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Request parameter values
-    // TODO: Check input values are filled (before allowing to submit)
-    UserService userService = UserServiceFactory.getUserService();
-    String email = userService.getCurrentUser().getEmail();
-    if (email == null) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid email.");
-      return;
+
+    // Retrieve body of HTTP request
+    StringBuffer requestBuffer = new StringBuffer();
+    try {
+      BufferedReader reader = request.getReader();
+      String currentLine;
+      while ((currentLine = reader.readLine()) != null) {
+        requestBuffer.append(currentLine);
+      }
+    } catch (Exception e) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Could not read request body");
     }
-    String username = email.split("@")[0];
+    JSONObject obj = new JSONObject(requestBuffer.toString());
```

Comment:
Looks like extracting information from this JSONObject can be in its own function, so it's easier to extend the fields. This doPost can be just taking results of that function call to create a new Participant. IMO, this doPost() should be the "main" function of adding a new participant, so it's the best to move as much as code out of it and instead use this function to call other functions.

Reply:
Thank you for the suggestion, I moved the JSON object extraction into its own function.